codec-http2/src/main/java/io/netty/handler/codec/http2/HpackHuffmanEncoder.java

```diff
@@ -87,7 +87,7 @@ private void encodeSlowPath(ByteBuf out, CharSequence data) {
         int n = 0;
 
         for (int i = 0; i < data.length(); i++) {
-            int b = data.charAt(i) & 0xFF;
+            int b = sanitizeChar(data.charAt(i));
```

Comment:
We cannot directly use `AsciiString.c2b(...)` here because this function does 2 things:

1. Sanitize the given input char.
2. Cast the sanitized char to a `byte`.

Then we cast the returned `byte` to `int b`. Overall, we are doing `sanitizedChar.toByte.toInt` and it could yield negative value which cannot be used as array index. For example `128.toByte.toInt == -128`.

Reply:
Updated it to `AsciiString.c2b(data.charAt(i)) & 0xFF` as `0xFF` can produce unsigned int which is what we want here.